File: LCA.java
public Node findLCA(Node root, int n1, int n2) {
    if (root == null)
        return null;

    if (root.data == n1 || root.data == n2)
        return root;

    Node leftLCA = findLCA(root.left, n1, n2);
    Node rightLCA = findLCA(root.right, n1, n2);

    if (leftLCA != null && rightLCA != null)
        return root;
    
    return (leftLCA != null) ? leftLCA : rightLCA;
}

File: LCA.cpp
Node* FindLCA(Node* root, int n1, int n2) {
    if (root == NULL)
        return NULL;

    if (root->data == n1 || root->data == n2)
        return root;

    Node* leftLCA = FindLCA(root->left, n1, n2);
    Node* rightLCA = FindLCA(root->right, n1, n2);

    if (leftLCA && rightLCA)
        return root;
    
    return (leftLCA != NULL) ? leftLCA : rightLCA;
}
